iaso/api/entity.py

```diff
@@ -325,69 +350,76 @@ def list(self, request: Request, *args, **kwargs):
                 }
                 result_list.append(result)
         else:
             for entity in entities:
-                etype_fields = entity.entity_type.fields_list_view
+                columns_list = []
                 entity_serialized = EntitySerializer(entity, many=False)
                 file_content = entity_serialized.data.get("attributes").get("file_content")
-                form_version = EntityType.objects.get(pk=entity_type_id).reference_form.latest_version
-                form_descriptor = form_version.get_or_save_form_descriptor()
-                for k in form_descriptor:
-                    form_key_list.append(k)
-                form_data_key = form_key_list[form_key_list.index("version") + 1]
-                descriptor_list = form_descriptor[form_data_key]
-
-                is_list = True
-                previous_name = None
-
-                # Get columns from xlsform file content
-                for d in descriptor_list:
-                    for k, v in d.items():
-                        data_list = v
-                        if k == "children":
-                            while is_list:
-                                for data in data_list:
-                                    value_dict = {}
-                                    for _k, _v in data.items():
-                                        if (
-                                            _k == "name"
-                                            and _v in etype_fields
-                                            or _k == "label"
-                                            and previous_name is not None
-                                            or _k == "type"
-                                            and previous_name is not None
-                                        ):
-                                            value_dict[_k] = _v
-                                            is_list = False
-                                            previous_name = _v
-                                        key_index = sorted(data.keys()).index(_k)
-                                        if key_index < len(sorted(data.keys())) - 1:
-                                            if sorted(data.keys())[key_index + 1] == "children":
-                                                data_list = data["children"]
-                                                is_list = True
-                                        if _k == "children":
-                                            data_list = _v
-                                            is_list = True
-                                    columns_list.append(value_dict)
-                result = {}
-                result["id"] = entity.pk
-                result["uuid"] = entity.uuid
-                result["entity_type_name"] = entity.entity_type.name
-                result["created_at"] = entity.created_at
-                result["updated_at"] = entity.updated_at
+
```

Comment:
General comment: the `list` method is getting really long and doing many different things. Would it be possible to 1) add a bit more comments (not what we do, but why we do it and describing more the API itself), and maybe split it into more digestable functions?